vsgenstubs4/_vapoursynth.part.pyi

```diff
@@ -516,236 +843,327 @@ class VideoFormat:
+
     def get_stride(self, plane: int) -> int: ...
 
     @property
-    def closed(self) -> bool: ...
+    def readonly(self) -> bool: ...
 
-    def close(self) -> None: ...
-    def __enter__(self) -> '_RawFrame': ...
-    def __exit__(self, ty: typing.Optional[typing.Type[BaseException]], tv: typing.Optional[BaseException], tb: typing.Optional[types.TracebackType]) -> None: ...
+    def __enter__(self: 'SelfFrame') -> 'SelfFrame': ...
 
+    def __exit__(
+        self, exc_type: Union[Type[BaseException], None],
+        exc_value: Union[BaseException, None],
+        traceback: Union[TracebackType, None], /,
+    ) -> Union[bool, None]: ...
 
-class VideoFrame(_RawFrame):
-    height: int
-    width: int
+    def __getitem__(self, index: int) -> memoryview: ...
+
+    def __len__(self) -> int: ...
+
+
+SelfFrame = TypeVar('SelfFrame', bound=RawFrame)
+
+
+class VideoFrame(RawFrame):
     format: VideoFormat
+    width: int
+    height: int
 
-    def copy(self) -> 'VideoFrame': ...
+    def _writelines(self, write: Callable[[bytes], None]) -> None: ...
 
-    def __getitem__(self, index: int) -> memoryview: ...
-    def __len__(self) -> int: ...
-    def __enter__(self) -> 'VideoFrame': ...
 
+class AudioFrame(RawFrame):
+    sample_type: SampleType
+    bits_per_sample: int
+    bytes_per_sample: int
+    channel_layout: int
+    num_channels: int
 
-class _Future(typing.Generic[T]):
-    def set_result(self, value: T) -> None: ...
-    def set_exception(self, exception: BaseException) -> None: ...
-    def result(self) -> T: ...
-    def exception(self) -> typing.Optional[typing.NoReturn]: ...
 
+#include <plugins/implementations>
 
-Func = typing.Callable[..., typing.Any]
 
+class RawNode:
+    def __init__(self) -> NoReturn: ...  # type: ignore[misc]
 
-class Plugin:
-    identifier: str
-    namespace: str
-    name: str
+    def get_frame(self, n: int) -> RawFrame: ...
 
-    def functions(self) -> typing.Iterator[Function]: ...
+    @overload
+    def get_frame_async(self, n: int, cb: None = None) -> _Future[RawFrame]: ...
 
-    # get_functions is deprecated
-    def get_functions(self) -> typing.Dict[str, str]: ...
-    # list_functions is deprecated
-    def list_functions(self) -> str: ...
+    @overload
+    def get_frame_async(self, n: int, cb: Callable[[Union[RawFrame, None], Union[Exception, None]], None]) -> None: ...
 
+    def frames(
+        self, prefetch: Union[int, None] = None, backlog: Union[int, None] = None, close: bool = False
+    ) -> Iterator[RawFrame]: ...
 
-class Function:
-    plugin: Plugin
-    name: str
-    signature: str
-    return_signature: str
+    def set_output(self, index: int = 0) -> None: ...
 
-    @property
-    def __signature__(self) -> inspect.Signature: ...
-    def __call__(self, *args: typing.Any, **kwargs: typing.Any) -> typing.Any: ...
+    def is_inspectable(self, version: Union[int, None] = None) -> bool: ...
 
+    if not TYPE_CHECKING:
```

Comment:
Is TYPE_CHECKING ever False?

Reply:
It's false when a code editor is searching for members, eg for autocompletion.